src/disco/pages/DiscoPane/index.js

```diff
@@ -18,36 +18,52 @@ import { NAVIGATION_CATEGORY } from 'disco/constants';
 import { makeQueryStringWithUTM } from 'disco/utils';
 import Addon from 'disco/components/Addon';
 import Button from 'ui/components/Button';
+import type { MozAddonManagerType } from 'core/addonManager';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+import type {
+  ReactRouterLocationType,
+  ReactRouterMatchType,
+} from 'core/types/router';
+import type { ExternalDiscoResultType } from 'disco/reducers/discoResults';
+import type { AppState } from 'disco/store';
 
 import './styles.scss';
 
-export class DiscoPaneBase extends React.Component {
-  static propTypes = {
-    AddonComponent: PropTypes.func,
-    dispatch: PropTypes.func.isRequired,
-    errorHandler: PropTypes.object.isRequired,
-    handleGlobalEvent: PropTypes.func.isRequired,
-    i18n: PropTypes.object.isRequired,
-    location: PropTypes.object.isRequired,
-    mozAddonManager: PropTypes.object,
-    match: PropTypes.shape({
-      params: PropTypes.shape({
-        platform: PropTypes.string.isRequired,
-      }).isRequired,
-    }).isRequired,
-    results: PropTypes.arrayOf(PropTypes.object).isRequired,
-    _addChangeListeners: PropTypes.func,
-    _tracking: PropTypes.object,
-  };
-
+type Props = {|
+  location: ReactRouterLocationType,
+  match: {|
```

Comment:
seems like these should be in InteralProps, no?

Reply:
Not here because this is a page-level component, which gets the `location` and `match` props from the `Route` component in the `Routes` component (if that makes sense :D)